Update DayTrader3 Dashboard to DayTrader7 Dashboard

dhlee49 opened the pull request:
- Removed Daytrader3 throughput dashboard 
- added Daytrader 7 startup dashboard for monitoring 
related to #118,#223
Signed-off-by: Dong Hoon Lee <l.donghoon49@gmail.com>

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+            </RangeSelector>
+
+            <Navigator>
+                <Navigator.Series seriesId="globalThroughput" />
```
Need to rename it since this is for startup now.

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+    "Test_openjdk8_j9_sanity.perf_x86-64_linux_Liberty": "Test_openjdk8_j9_sanity.perf_x86-64_linux_Liberty",
+};
+let display = {
+    "Test_openjdk8_j9_sanity.perf_x86-64_linux_Libertym": true,
```
Typo: `_Libertym`

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+    "Test_openjdk8_j9_sanity.perf_x86-64_linux_Libertym": true,
+};
+//currently no baseline runs are made we use hard coded baseline values for scaling graph
+let baseLineValue = 300000;
```
To be consistent, it should be `baselineValue`. Similar issue in other places as well.

llxia replied:
If the value does not change, please use `const` not `let`. Please fix this issue for the rest of file

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+//currently no baseline runs are made we use hard coded baseline values for scaling graph
+let baseLineValue = 300000;
+let scale = baseLineValue/100;
+let baseLineValuesu = 13000;
```
Need a better to name to reflect what it holds.

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+let scale = baseLineValue/100;
+let baseLineValuesu = 13000;
+let scalesu = baseLineValuesu/100;
+export class DayTrader3Setting extends Component {
```
Let's update all `DayTrader3` instances with `DayTrader7` everywhere.

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+        } );
+        const results = await response.json();
+        const startupresultsByJDKBuild = {};
+        const footprintresultsByJDKBuild = {};
```
Since we're using camel case, `startupresultsByJDKBuild` should be `startupResultsByJDKBuild`. Same thing for other variables.

llxia commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+                    footprintresultsByJDKBuild[t.buildName] = {};
+                }
+                footprintresultsByJDKBuild[t.buildName][jdkDate] = footprintresultsByJDKBuild[jdkDate] || [];
+                footprintresultsByJDKBuild[t.buildName][jdkDate].push( {
```
Why we need to add `element.metrics[0]` and `element.metrics[1]` specifically? Can we use a loop here?

llxia commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -0,0 +1,249 @@
+
+    formatter = function() {
+        const x = new Date( this.x );
+        const CIstr = (typeof this.point.CI === 'undefined') ? ``: `CI = ${this.point.CI}`;
```
const CIstr = this.point.CI ? `CI = ${this.point.CI}` : "";

dhlee49 commented:
Updated PR mostly 
1.Fix the minor typos & wrong terms
2.Updated metrics into single loop
regard https://github.com/AdoptOpenJDK/openjdk-test-tools/pull/233#discussion_r413272716, currently Liberty runs 2 times a day so we need precise timestamp (as jdkDate provides only date the data collapses together on 1 point which I believe we are not intending to)

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -1,195 +1,218 @@
+                    if(!resultsByJDKBuild[t.buildName]) {
+                        resultsByJDKBuild[t.buildName] = {} ;
+                    }
+                    for( let singleMetric of element.metrics) {
```
Might be better to shorten it to just `metric` instead of `singleMetric`.

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -1,195 +1,218 @@
+        } );
+        let baseLineData = [];
+        let metric_data = [];
+        for ( let singleTestName in resultsByJDKBuild) {
```
`singleTestName` isn't reflective of what it stores. I think it should be something like `buildName`.

piyush286 commented on test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx:
```diff
@@ -1,195 +1,218 @@
+            if (!metric_data[singleTestName]){
+                metric_data[singleTestName] = [];
+            }
+            for ( let singleMetricName in resultsByJDKBuild[singleTestName]){
```
Same thing as before. Maybe just call it `metric` instead of `singleMetricName`.

piyush286 commented:
Posting screenshots here for reference: 
![image](https://user-images.githubusercontent.com/10422081/81025348-657f8e00-8e44-11ea-8740-8f951a968e11.png)

![image](https://user-images.githubusercontent.com/10422081/81025371-73351380-8e44-11ea-9d66-c6e36ce3215b.png)

dhlee49 commented:
updated PR ( sorry about 3 push made errors in the first 2)
1.renaming
2.timestamp to JDKdate
3.convert let to const for ones needed
4. comments 
Below is graph generated locally from above PR
<img width="905" alt="Screen Shot 2020-05-06 at 9 11 44 AM" src="https://user-images.githubusercontent.com/22691165/81180891-a5945d00-8f79-11ea-9cea-e17bf7e929dd.png">
